rfcs/20241205-graph-api-causal-mask/README.md

```diff
@@ -0,0 +1,417 @@
+# Support Implicit Causal Mask in SDPA in Graph API
+
+## Background
+
+The causal mask is an essential component in Scaled Dot-Product Attention (SDPA), particularly for autoregressive language models like GPT. Its primary purpose is to prevent information leakage from future tokens during the training and inference processes of language models. To achieve this, causal mask is typically implemented as an upper triangular matrix with zeros on and below the main diagonal, and negative infinity above it [1]. This configuration ensures that each token can only attend to itself and the tokens that come before it in the sequence.
```

Comment:
Thanks for the RFC.

nit: better to wrap the lines at 80 characters. Same for the below sections.

Reply:
Thanks for the reminder, fixed.